src/v2/Utils/getSaleOrLotTimerInfo.ts

```diff
@@ -0,0 +1,91 @@
+export interface Time {
+  days: string
+  hours: string
+  minutes: string
+  seconds: string
+}
+
+interface TimerInfo {
+  copy: string
+  color: string
+}
+
+// If `extendedBiddingEndAt` is passed in and non-null, the `time`
+// parameter refers to the time left in the extended period.
+export const getSaleOrLotTimerInfo = (
+  time: Time,
+  hasStarted?: boolean,
+  saleHasEnded?: boolean,
+  isSaleInfo: boolean = false,
+  extendedBiddingEndAt?: string | null
+): TimerInfo => {
+  const { days, hours, minutes, seconds } = time
+
+  const parsedDays = parseInt(days, 10)
+  const parsedHours = parseInt(hours, 10)
+  const parsedMinutes = parseInt(minutes, 10)
+  const parsedSeconds = parseInt(seconds, 10)
+
+  let copy = ""
+  let color = "blue100"
+
+  // Sale has not yet started
+  if (!hasStarted) {
+    if (parsedDays < 1) {
+      copy = "Bidding Starts Today"
+      // Entered extended bidding
+    } else if (extendedBiddingEndAt) {
+      copy = `Extended: ${parsedMinutes}m ${parsedSeconds}s`
+      color = "red100"
+    } else {
+      copy = `${parsedDays} Day${
+        parsedDays > 1 ? "s" : ""
+      } Until Bidding Starts`
+    }
+    // Sale has started
+  } else {
+    // When the timer is on the sale:
+    if (isSaleInfo) {
+      if (saleHasEnded) {
```

Comment:
This line and the next kind of read oddly, if the sale has ended display 'Lots are closing'? I think in the use-case of a cascading sale, the `saleHasEnded` parameter _means_ that the lots have started closing. I wonder if that can be captured better via the grab-bag of parameters being passed in or parameter naming.

Reply:
I believe the sale and artwork page usage of this , so in general, I think this component is only used when a sale has cascading enabled, so maybe the argument should be renamed to `isLotsClosing` or something like that.

Reply:
Yeah `lotsAreClosing` is much better and I've used that elsewhere for this reason. I'll use that instead